File: Data Structures & Algorithms/three-integer-sum/submission-0.py
class Solution:
    def threeSum(self, nums: List[int]) -> List[List[int]]:
        res = []
        nums.sort()
        for i, num in enumerate(nums):
            if num > 0:
                break
            if i > 0 and num == nums[i - 1]:  # skip duplicate num
                continue
            l, r = i+1, len(nums)-1
            while l < r:
                if num + nums[l] + nums[r] > 0:
                    r -= 1
                elif num + nums[l] + nums[r] < 0:
                    l += 1
                else:
                    res.append([num, nums[l], nums[r]])

                    while l < r and nums[l] == nums[l + 1]:
                        l += 1
                    # skip duplicates for nums[r]
                    while l < r and nums[r] == nums[r - 1]:
                        r -= 1
                    l += 1
                    r -= 1

        return res

File: Data Structures & Algorithms/height-checker/submission-0.cpp
class Solution {
public:
    int heightChecker(vector<int>& heights) {
        vector<int> h = heights;
        sort(h.begin(), h.end());
        int res = 0;
        for (int i= 0; i < heights.size(); i++){
            if (h[i] != heights[i]){
                res++;
            } 
        }
        return res;
    }
};